src/sagemaker/workflow/clarify_check_step.py

```diff
@@ -266,8 +269,13 @@ def arguments(self) -> RequestType:
         request_dict = self._baselining_processor.sagemaker_session._get_process_request(
             **process_args
         )
-        if "ProcessingJobName" in request_dict:
-            request_dict.pop("ProcessingJobName")
+        # Continue to pop custom prefix if not explicitly opted-in
+        if not _pipeline_config or not _pipeline_config.use_custom_job_prefix:
+            if "ProcessingJobName" in request_dict:
```

Comment:
Is there a reason this is the only place where the key is checked before calling `.pop`? Do we want a KeyError thrown for the other cases and can we add unit tests for this scenario?

Reply:
Will be worked into fix for comment above.

Reply:
Most other steps take a default on returning `None` to avoid this case in which we can treat it as a NOOP. In my opinion either way is fine, the latter being a bit more concise. [[link]](https://www.javatpoint.com/python-dictionary-pop-method#:~:text=Python%20pop()%20method%20removes,it%20throws%20an%20error%20KeyError.)